core/api/wallet/local_wallet.cpp

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "api/wallet/local_wallet.hpp"
+
+#include "common/error_text.hpp"
+#include "crypto/bls/impl/bls_provider_impl.hpp"
+#include "crypto/secp256k1/impl/secp256k1_provider_impl.hpp"
+#include "vm/state/impl/state_tree_impl.hpp"
+#include "vm/state/resolve_key.hpp"
+
+namespace fc::api {
+  using vm::state::StateTreeImpl;
+
+  void LocalWallet::fillLocalWalletApi(
+      const std::shared_ptr<WalletApi> &api,
+      const std::shared_ptr<KeyStore> &key_store,
+      const IpldPtr &ipld,
+      const std::shared_ptr<OneKey> &wallet_default_address) {
+    const StateTreeImpl state_tree{ipld};
```

Comment:
```suggestion
      const std::function<outcome::result<TipsetContext>()> &get_ts_ctx,
      const std::shared_ptr<OneKey> &wallet_default_address) {
```
can move `tipsetContext` lambda from `api/full_node/make.cpp` to `builder.cpp` and reuse for both/all `makeImpl/fill...`